namastack-outbox-core/src/main/kotlin/io/namastack/outbox/handler/method/handler/TypedHandlerMethod.kt

```diff
@@ -20,10 +21,16 @@ class TypedHandlerMethod(
         get() = method.parameterTypes.first().kotlin
 
     /**
-     * Invokes handler with typed payload via reflection.
+     * Invokes handler with typed payload and metadata.
      *
      * @param payload Record payload matching paramType
+     * @param metadata Record metadata
      * @throws Throwable Original exception from handler (triggers retry logic)
      */
-    fun invoke(payload: Any) = invokeMethod(payload)
+    fun invoke(
+        payload: Any,
+        metadata: OutboxRecordMetadata,
+    ) {
+        invokeMethod(payload, metadata)
```

Comment:
I am unsure what your plan is for the `OutboxHandler` annotation, so the comment may be unnecessary. The metadata parameter in annotation-based handlers should be optional, allowing users to decide whether they want to include this argument or not.
```kotlin
@OutboxHandler
fun handle(payload: OrderCreatedEvent) {
    ...
}

@OutboxHandler
fun handle(payload: OrderCreatedEvent, matadata: OutboxRecordMetadata) {
    ...
}
```